### PR TITLE
SAIL-16 Sync Product Data Nightly to Sailthru

### PR DESCRIPTION
**Pull request by ladonenko**

### CODE REVIEW

File: Model/Cron.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        $collectionSize = $collection->getSize();
+        $attemptsCount = $this->flagManager->getFlagData(self::FLAG_NAME);
+        if (($attemptsCount === null) || ($attemptsCount <= 0)) {
+            $attemptsCount = 0;
+        }
+        for ($i = $attemptsCount; $i < $collectionSize; $i += self::LOAD_COLLECTION_STEP) {
+            $collection
+                ->addAttributeToSelect('*')
+                ->setOrder('entity_id', 'ASC')
+                ->addFieldToFilter('entity_id', ['from' => $i, 'to' => $i + self::LOAD_COLLECTION_STEP]);;
```
Comment (giraclion) [inline Model/Cron.php]:
```
$collection->getSelect()
            ->order('entity_id' . Varien_Data_Collection::SORT_ORDER_ASC)
            ->limit($limit);
```

File: Model/Cron.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        foreach ($storeManagerDataList as $store) {
+            $storesCronStatus[$store->getStoreId()] = (int)$this->sailthruProduct->isSyncProductCronEnable($store->getStoreId());
+        }
+        if(!in_array(1, $storesCronStatus)) {
+            return false;
+        }
+
+        $collection = $this->collectionFactory->create();
+        $collectionSize = $collection->getSize();
+        $attemptsCount = $this->flagManager->getFlagData(self::FLAG_NAME);
```
Comment (giraclion) [inline Model/Cron.php]:
```
$collectionLastEntityId = $this->flagManager->getFlag(self::FLAG_NAME);
for ($i = 0; $i <= $colelctionSize; $ += STEP_VALUE) {
     if $collectionLastEntityId NOT EMPTY
         SELECT * FROM orders WHERE entity_id > $collectionLastEntityId LIMIT STEP_VALUE;
     // process collection
     $this->flagManager->saveFlag(self::FLAG_NAME, $collectionLastEntityId);
}

```

File: Model/Cron.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    public function exportProducts()
+    {
+
+        $storeManagerDataList = $this->_storeManager->getStores();
+        $storesCronStatus = [];
+        $storesCronStatus[0] = (int)$this->sailthruProduct->isSyncProductCronEnable();
+        foreach ($storeManagerDataList as $store) {
+            $storesCronStatus[$store->getStoreId()] = (int)$this->sailthruProduct->isSyncProductCronEnable($store->getStoreId());
+        }
+        if(!in_array(1, $storesCronStatus)) {
```
Comment (giraclion) [inline Model/Cron.php]: Move `1` to constant and update code style

File: Model/Cron.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        $collection = $this->collectionFactory->create();
+        $collectionSize = $collection->getSize();
+        $attemptsCount = $this->flagManager->getFlagData(self::FLAG_NAME);
+        if (($attemptsCount === null) || ($attemptsCount <= 0)) {
+            $attemptsCount = 0;
+        }
+        for ($i = $attemptsCount; $i < $collectionSize; $i += self::LOAD_COLLECTION_STEP) {
+            $collection
+                ->addAttributeToSelect('*')
+                ->setOrder('entity_id', 'ASC')
```
Comment (giraclion) [inline Model/Cron.php]: Use constant `SORT_ORDER_ASC` instead of `'ASC'`

**Comment by ladonenko** [comment]
Fixed

**Comment by ladonenko** [comment]
fixed

**Pull request merged**